t.geoserver.publish/t.geoserver.publish.py

```diff
@@ -44,6 +44,7 @@
 # % type: string
 # % required: no
 # % multiple: no
+# % options: bcyr,bgyr,blues,byg,byr,default,elevation,evi,forest_cover,grass,greens,grey,gyr,ndvi,ndwi,reds,ryb,ryg
 # % label: Name of color table for layer styling
 # %end
```

Comment:
Please consider to use the related macro

`# %option G_OPT_M_COLR`

see e.g.:

https://github.com/OSGeo/grass/blob/e1756b57d61affa0521ad834590ca2245f9c0e9a/temporal/t.rast.colors/t.rast.colors.py#L34

Reply:
Thanks! I add the `default`- answer, plus I limit the allowed options. Is this valid when using the macro? (probably yes as it only overwrites parts of the macro?)

Reply:
After testing, I would refrain from using `# %option G_OPT_M_COLR`, as together with using `options` it yields unclear Warnings when calling the module:
```
GRASS utm32n_wgs84/openeo_example_strds:~ > t.geoserver.publish input=S2_STRDS_B04_10m mosaic_layername=bla color=default
WARNUNG: Bug in UI description. Option 'aspect' in <color> does not exist
WARNUNG: Bug in UI description. Option 'aspectcolr' in <color> does not
         exist
WARNUNG: Bug in UI description. Option 'celsius' in <color> does not exist
WARNUNG: Bug in UI description. Option 'corine' in <color> does not exist
WARNUNG: Bug in UI description. Option 'curvature' in <color> does not
         exist
WARNUNG: Bug in UI description. Option 'differences' in <color> does not
         exist
WARNUNG: Bug in UI description. Option 'etopo2' in <color> does not exist
WARNUNG: Bug in UI description. Option 'fahrenheit' in <color> does not
         exist
WARNUNG: Bug in UI description. Option 'gdd' in <color> does not exist
WARNUNG: Bug in UI description. Option 'grey.eq' in <color> does not exist
WARNUNG: Bug in UI description. Option 'grey.log' in <color> does not exist
WARNUNG: Bug in UI description. Option 'grey1.0' in <color> does not exist
WARNUNG: Bug in UI description. Option 'grey255' in <color> does not exist
WARNUNG: Bug in UI description. Option 'haxby' in <color> does not exist
WARNUNG: Bug in UI description. Option 'inferno' in <color> does not exist
WARNUNG: Bug in UI description. Option 'kelvin' in <color> does not exist
WARNUNG: Bug in UI description. Option 'magma' in <color> does not exist
WARNUNG: Bug in UI description. Option 'nlcd' in <color> does not exist
WARNUNG: Bug in UI description. Option 'oranges' in <color> does not exist
WARNUNG: Bug in UI description. Option 'plasma' in <color> does not exist
WARNUNG: Bug in UI description. Option 'population' in <color> does not
         exist
WARNUNG: Bug in UI description. Option 'population_dens' in <color> does
         not exist
WARNUNG: Bug in UI description. Option 'precipitation' in <color> does not
         exist
WARNUNG: Bug in UI description. Option 'precipitation_daily' in <color>
         does not exist
WARNUNG: Bug in UI description. Option 'precipitation_monthly' in <color>
         does not exist
WARNUNG: Bug in UI description. Option 'rainbow' in <color> does not exist
WARNUNG: Bug in UI description. Option 'ramp' in <color> does not exist
WARNUNG: Bug in UI description. Option 'random' in <color> does not exist
WARNUNG: Bug in UI description. Option 'roygbiv' in <color> does not exist
WARNUNG: Bug in UI description. Option 'rstcurv' in <color> does not exist
WARNUNG: Bug in UI description. Option 'sepia' in <color> does not exist
WARNUNG: Bug in UI description. Option 'slope' in <color> does not exist
WARNUNG: Bug in UI description. Option 'soilmoisture' in <color> does not
         exist
WARNUNG: Bug in UI description. Option 'srtm' in <color> does not exist
WARNUNG: Bug in UI description. Option 'srtm_plus' in <color> does not
         exist
WARNUNG: Bug in UI description. Option 'terrain' in <color> does not exist
WARNUNG: Bug in UI description. Option 'viridis' in <color> does not exist
WARNUNG: Bug in UI description. Option 'water' in <color> does not exist
WARNUNG: Bug in UI description. Option 'wave' in <color> does not exist

```

Reply:
Aren't the color names listed as `... does not exist` those color names which you removed from the full list by defining a subset?
Seems the parser isn't prepared for that (yet).

Reply:
yes indeed. I meant they are "unclear" such that a user might not know what to do with this warning at all.